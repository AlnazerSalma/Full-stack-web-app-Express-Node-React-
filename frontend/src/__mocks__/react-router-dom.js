import React from "react";

export const MemoryRouter = ({ children }) => children;
export const useLocation = () => ({ pathname: "/" });
export const Link = ({ to, children }) => <a href={to}>{children}</a>;
