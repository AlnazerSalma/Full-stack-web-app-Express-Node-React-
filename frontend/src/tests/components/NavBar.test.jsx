import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import NavBar from "../../components/Navbar";
import { MemoryRouter } from "react-router-dom";

// Utility for rendering NavBar with a custom route
const renderWithRouter = (ui, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);
  return render(<MemoryRouter initialEntries={[route]}>{ui}</MemoryRouter>);
};

describe("NavBar Component", () => {
  it("renders logo correctly", () => {
    renderWithRouter(<NavBar />);
    const logo = screen.getByAltText("brand");
    expect(logo).toBeInTheDocument();
  });

  it("renders all navigation links", () => {
    renderWithRouter(<NavBar />);
    const expectedLinks = ["Home", "Work", "Pricing", "Careers", "Contact"];
    expectedLinks.forEach((text) => {
      expect(screen.getByText(text)).toBeInTheDocument();
    });
  });
  it("nav links have correct href attribute matching the route path", () => {
    renderWithRouter(<NavBar />);
    const paths = ["/", "/work", "/pricing", "/careers", "/contact"];
    paths.forEach((path) => {
      const linkText = path === "/" ? "Home" : path.charAt(1).toUpperCase() + path.slice(2);
      const link = screen.getByText(linkText);
      expect(link).toHaveAttribute("href", path);
    });
  });
  
  it("toggles navbar on hamburger click", () => {
    renderWithRouter(<NavBar />);
    const toggleButton = screen.getByLabelText("Toggle navigation");
    fireEvent.click(toggleButton);
    // Ensure no crash, menu still visible
    expect(screen.getByText("Home")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<NavBar />);
  expect(asFragment()).toMatchSnapshot();
});

});

