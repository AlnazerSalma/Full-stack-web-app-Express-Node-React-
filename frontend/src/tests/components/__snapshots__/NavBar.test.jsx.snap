// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md navbar-light fixed-top"
  >
    <div
      class="container"
    >
      <a
        class="d-flex navbar-brand"
        href="/"
      >
        <img
          alt="brand"
          class="img-fluid logo"
          src="logo.png"
        />
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span />
        <span />
        <span />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="ms-auto navbar-nav"
        >
          <div
            class="nav-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              href="/work"
            >
              Work
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              href="/pricing"
            >
              Pricing
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              href="/careers"
            >
              Careers
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
